utils/automation/FoScanner/Utils.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+
+# SPDX-FileContributor: Rajul Jha <rajuljha49@gmail.com>
+# SPDX-License-Identifier: GPL-2.0-only
+
+import os
+import re
+import shutil
+
+def validate_keyword_conf_file(file_path:str):
+    """
+    Validate whether input file in keyword.conf format or not
+    :param: file_path : str File path to vaildate
+    :return: bool True if file in correct format. Else False
+    :return: str Validation message(True)/Error message(False)
+    """
+    try:
+      with open(file_path, 'r') as file:
+        lines = file.readlines()
+      if not lines:
+        return False, "File is empty"
+      commented_lines = []
+      non_commented_lines = []
+      for line in lines:
+        stripped_line = line.strip()
+        if stripped_line.startswith('#') or not stripped_line:
+          commented_lines.append(stripped_line)
+        else:
+          non_commented_lines.append(stripped_line)
+
+      if not non_commented_lines:
+        return False, "File has no keyword to seach for"
+
+      # Check for the SPACESALL definition on the first non-commented line
+      spacesall_pattern = re.compile(r'SPACESALL=\[\[:space:\]\]\*')
+      if not spacesall_pattern.match(non_commented_lines[0]):
+        return False, "First non-commented line must define SPACESALL as [[:space:]]*"
+
+      # Check each subsequent non-commented line for the keyword pattern
+      keyword_pattern = re.compile(r'keyword=__keyword__\|?\(.*\)')
+      for line in non_commented_lines[1:]:
+        if not keyword_pattern.match(line):
+          return False, f"Invalid keyword pattern: {line}"
```

Comment:
There is no strict requirement to define `SPACESALL` in the conf file. All what it should contain is at least a `keyword=` line.

And the special symbol `__` must always occur in pair (if it exists).

And the non-comment lines must contain only 1 `=`

Reply:
Modified it accordingly.
Please let me know if I missed something.